fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -1470,6 +1468,67 @@ private void checkReplicaContinuousVersionSucc(long tabletId, Replica replica, b
         }
     }
 
+    private void checkReplicaContinuousVersionSucc(long tabletId, Replica replica, boolean alterReplicaLoadedTxn,
+            long minReplicaVersion, long maxReplicaVersion, List<PublishVersionTask> replicaPublishTasks,
+            Set<Long> errorReplicaIds, List<Replica> tabletSuccReplicas, List<Replica> tabletWriteFailedReplicas,
+            List<Replica> tabletVersionFailedReplicas) {
+        boolean failed = false;
+        for (PublishVersionTask task : replicaPublishTasks) {
+            if (task == null || !task.isFinished()) {
+                errorReplicaIds.add(replica.getId());
+                failed = true;
+            }
+        }
+        if (!failed) {
+            for (PublishVersionTask task : replicaPublishTasks) {
+                Map<Long, Long> backendSuccTablets = task.getSuccTablets();
```

Comment:
for (int i=0; i<subTxnIds.size(); i++) {
            PublishVersionTask task = replicaPublishTasks.get(i);
            boolean succ = task != null && task.isFinished() && task.getSuccTablets().contains(tabletId);
                      || (replica.getState() == Replica.ReplicaState.ALTER
                                  && (!Config.publish_version_check_alter_replica || (alterTxnWatermark > 0 && subTxnIds.get(i) < alterTxnWatermark)));

}